app/models/vm_or_template.rb

```diff
@@ -194,17 +197,18 @@ class VmOrTemplate < ApplicationRecord
   before_validation :set_tenant_from_group
   after_save :save_genealogy_information
 
-  scope :active,       ->       { where.not(:ems_id => nil) }
-  scope :with_type,    ->(type) { where(:type => type) }
-  scope :archived,     ->       { where(:ems_id => nil, :storage_id => nil) }
-  scope :orphaned,     ->       { where(:ems_id => nil).where.not(:storage_id => nil) }
-  scope :retired,      ->       { where(:retired => true) }
-  scope :with_ems,     ->       { where.not(:ems_id => nil) }
-  scope :not_archived, ->       { where.not(:ems_id => nil).or(where.not(:storage_id => nil)) }
-  scope :not_orphaned, ->       { where.not(:ems_id => nil).or(where(:storage_id => nil)) }
-  scope :not_retired,  ->       { where(:retired => false).or(where(:retired => nil)) }
-
-  # The SQL form of `#registered?`, with it's inverse as well.
+  scope :active,       ->         { where.not(:ems_id => nil) }
+  scope :with_type,    ->(type)   { where(:type => type) }
+  scope :archived,     ->         { where(:ems_id => nil, :storage_id => nil) }
+  scope :orphaned,     ->         { where(:ems_id => nil).where.not(:storage_id => nil) }
+  scope :retired,      ->         { where(:retired => true) }
+  scope :with_ems,     ->         { where.not(:ems_id => nil) }
+  scope :not_archived, ->         { where.not(:ems_id => nil).or(where.not(:storage_id => nil)) }
+  scope :not_orphaned, ->         { where.not(:ems_id => nil).or(where(:storage_id => nil)) }
+  scope :not_retired,  ->         { where(:retired => false).or(where(:retired => nil)) }
+  scope :by_ems,       ->(ems_id) { includes(:ext_management_system).where(:ems_id => ems_id).references(:ext_management_system) }
```

Comment:
I think that when you use `ems_id`
you don't want includes and references here:

1. 1 query
```ruby
irb(main):035:0> VmOrTemplate.where(:ems_id => ems_id).first

  VmOrTemplate Load (1.3ms)  SELECT  "vms".* FROM "vms" WHERE "vms"."ems_id" = $1 ORDER BY "vms"."id" ASC LIMIT $2  [["ems_id", 72000000000001], ["LIMIT", 1]]
```

2.2 queries
```ruby
irb(main):035:0> VmOrTemplate.includes(:ext_management_system).where(:ems_id => ems_id).first
  
VmOrTemplate Load (1.1ms)  SELECT  "vms".* FROM "vms" WHERE "vms"."ems_id" = $1 ORDER BY "vms"."id" ASC 
LIMIT $2  [["ems_id", 72000000000001], ["LIMIT", 1]]
  VmOrTemplate Inst Including Associations (0.3ms - 1rows)
  ExtManagementSystem Load (0.4ms)  SELECT "ext_management_systems".* FROM "ext_management_systems" WHERE "ext_management_systems"."id" = 72000000000001
  ExtManagementSystem Inst Including Associations (0.3ms - 1rows)
```
3. 1 query - but unnecessarily  complex.
```ruby
VmOrTemplate.includes(:ext_management_system).where(:ems_id => ems_id).references(:ext_management_system).first
  SQL (2.4ms)  SELECT  "vms"."id" AS t0_r0, "vms"."vendor" AS t0_r1, "vms"."format" AS t0_r2, "vms"."version" AS t0_r3, "vms"."name" AS t0_r4, "vms"."description" AS t0_r5, "vms"."location" AS t0_r6, "vms"."config_xml" AS t0_r7, "vms"."autostart" AS t0_r8, "vms"."host_id" AS t0_r9, "vms"."last_sync_on" AS t0_r10, "vms"."created_on" AS t0_r11, "vms"."updated_on" AS t0_r12, "vms"."storage_id" AS t0_r13, "vms"."guid" AS t0_r14, "vms"."ems_id" AS t0_r15, "vms"."last_scan_on" AS t0_r16, "vms"."last_scan_attempt_on" AS t0_r17, "vms"."uid_ems" AS t0_r18, "vms"."retires_on" AS t0_r19, "vms"."retired" AS t0_r20, "vms"."boot_time" AS t0_r21, "vms"."tools_status" AS t0_r22, "vms"."standby_action" AS t0_r23, "vms"."power_state" AS t0_r24, "vms"."state_changed_on" AS t0_r25, "vms"."previous_state" AS t0_r26, "vms"."connection_state" AS t0_r27, "vms"."last_perf_capture_on" AS t0_r28, "vms"."registered" AS t0_r29, "vms"."busy" AS t0_r30, "vms"."smart" AS t0_r31, "vms"."memory_reserve" AS t0_r32, "vms"."memory_reserve_expand" AS t0_r33, "vms"."memory_limit" AS t0_r34, "vms"."memory_shares" AS t0_r35, "vms"."memory_shares_level" AS t0_r36, "vms"."cpu_reserve" AS t0_r37, "vms"."cpu_reserve_expand" AS t0_r38, "vms"."cpu_limit" AS t0_r39, "vms"."cpu_shares" AS t0_r40, "vms"."cpu_shares_level" AS t0_r41, "vms"."cpu_affinity" AS t0_r42, "vms"."ems_created_on" AS t0_r43, "vms"."template" AS t0_r44, "vms"."evm_owner_id" AS t0_r45, "vms"."ems_ref_obj" AS t0_r46, "vms"."miq_group_id" AS t0_r47, "vms"."linked_clone" AS t0_r48, "vms"."fault_tolerance" AS t0_r49, "vms"."type" AS t0_r50, "vms"."ems_ref" AS t0_r51, "vms"."ems_cluster_id" AS t0_r52, "vms"."retirement_warn" AS t0_r53, "vms"."retirement_last_warn" AS t0_r54, "vms"."vnc_port" AS t0_r55, "vms"."flavor_id" AS t0_r56, "vms"."availability_zone_id" AS t0_r57, "vms"."cloud" AS t0_r58, "vms"."retirement_state" AS t0_r59, "vms"."cloud_network_id" AS t0_r60, "vms"."cloud_subnet_id" AS t0_r61, "vms"."cloud_tenant_id" AS t0_r62, "vms"."raw_power_state" AS t0_r63, "vms"."publicly_available" AS t0_r64, "vms"."orchestration_stack_id" AS t0_r65, "vms"."retirement_requester" AS t0_r66, "vms"."tenant_id" AS t0_r67, "vms"."resource_group_id" AS t0_r68, "vms"."deprecated" AS t0_r69, "vms"."storage_profile_id" AS t0_r70, "vms"."cpu_hot_add_enabled" AS t0_r71, "vms"."cpu_hot_remove_enabled" AS t0_r72, "vms"."memory_hot_add_enabled" AS t0_r73, "vms"."memory_hot_add_limit" AS t0_r74, "vms"."memory_hot_add_increment" AS t0_r75, "vms"."hostname" AS t0_r76, "ext_management_systems"."id" AS t1_r0, "ext_management_systems"."name" AS t1_r1, "ext_management_systems"."created_on" AS t1_r2, "ext_management_systems"."updated_on" AS t1_r3, "ext_management_systems"."guid" AS t1_r4, "ext_management_systems"."zone_id" AS t1_r5, "ext_management_systems"."type" AS t1_r6, "ext_management_systems"."api_version" AS t1_r7, "ext_management_systems"."uid_ems" AS t1_r8, "ext_management_systems"."host_default_vnc_port_start" AS t1_r9, "ext_management_systems"."host_default_vnc_port_end" AS t1_r10, "ext_management_systems"."provider_region" AS t1_r11, "ext_management_systems"."last_refresh_error" AS t1_r12, "ext_management_systems"."last_refresh_date" AS t1_r13, "ext_management_systems"."provider_id" AS t1_r14, "ext_management_systems"."realm" AS t1_r15, "ext_management_systems"."tenant_id" AS t1_r16, "ext_management_systems"."project" AS t1_r17, "ext_management_systems"."parent_ems_id" AS t1_r18, "ext_management_systems"."subscription" AS t1_r19, "ext_management_systems"."last_metrics_error" AS t1_r20, "ext_management_systems"."last_metrics_update_date" AS t1_r21, "ext_management_systems"."last_metrics_success_date" AS t1_r22, "ext_management_systems"."tenant_mapping_enabled" AS t1_r23, "ext_management_systems"."enabled" AS t1_r24, "ext_management_systems"."options" AS t1_r25, "ext_management_systems"."zone_before_pause_id" AS t1_r26, "ext_management_systems"."last_inventory_date" AS t1_r27 FROM "vms" LEFT OUTER JOIN "ext_management_systems" ON "ext_management_systems"."id" = "vms"."ems_id" WHERE "vms"."ems_id" = $1 ORDER BY "vms"."id" ASC LIMIT $2  [["ems_id", 72000000000001], ["LIMIT", 1]]
```

so I think that we should use  1) 

cc @kbrock